sdk/data/azcosmos/cosmos_global_endpoint_manager_test.go

```diff
@@ -0,0 +1,378 @@
+	assert.NoError(t, err)
+
+	// Assert the expected write endpoints after the update
+	expectedWriteEndpoints := []url.URL{
+		{Scheme: "https", Host: "westus.documents.azure.com"},
+		{Scheme: "https", Host: "eastus.documents.azure.com"},
+	}
+	assert.Equal(t, expectedWriteEndpoints, writeEndpoints)
+
+	// Get the read endpoints after the update
+	readEndpoints, err := gem.GetReadEndpoints()
+	assert.NoError(t, err)
+
+	// Assert the expected read endpoints after the update
+	expectedReadEndpoints := []url.URL{
+		{Scheme: "https", Host: "westus.documents.azure.com"},
+		{Scheme: "https", Host: "eastus.documents.azure.com"},
+	}
+	assert.Equal(t, expectedReadEndpoints, readEndpoints)
+}
+
+func TestGlobalEndpointManager_RefreshStaleEndpoints(t *testing.T) {
+	// Create a mock client
+	// Create a mock client
+	headerPolicy := &headerPolicies{}
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	verifier := headerPoliciesVerify{}
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{headerPolicy, &verifier}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		return
+	}
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+	mockClient := client
+
+	// Create a new global endpoint manager with preferred regions and refresh interval
+	gem, err := newGlobalEndpointManager(mockClient, []string{"westus", "eastus"}, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Refresh stale endpoints in the location cache
+	gem.RefreshStaleEndpoints()
+	// No assertions as it is an internal mechanism
+}
+
+func TestGlobalEndpointManager_CanUseMultipleWriteLocations(t *testing.T) {
+	// Create a mock client
+	headerPolicy := &headerPolicies{}
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	verifier := headerPoliciesVerify{}
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{headerPolicy, &verifier}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		return
+	}
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+	mockClient := client
+
+	// Create a new global endpoint manager with preferred regions and refresh interval
+	gem, err := newGlobalEndpointManager(mockClient, []string{"westus", "eastus"}, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Check if multiple write locations can be used
+	canUseMultipleWriteLocs := gem.CanUseMultipleWriteLocations()
+	assert.False(t, canUseMultipleWriteLocs)
+}
+
+func TestGlobalEndpointManagerBackgroundRefresh(t *testing.T) {
```

Comment:
Is the test expected to validate that backgroundRefresh is up and running in background and doing account refreshes? 

If so the test needs to validate below
- background refresh continuing even after current refresh failure
- Between retries refreshInterval